### ISSUE TITLE
Add BUILD files for the rest of the C++ targets and tests. [phab:D213]

### ISSUE THREAD

**Issue by kythe** [google/kythe#425]
_Created by **zarko** at 2015-04-22 23:26:37_: 

TODO(zarko): factor out a common test init script
from the various sh_tests to set binary/output paths.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-22 23:33:54_:

  - This one gets me every time (revert grpc BUILD)

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-23 16:44:58_:

We should have TODOs to remove the campfire-specific workarounds.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-23 16:44:58_:

`//kythe/extractors/cmake:compilation_database` fails with this log:

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
/home/schroederc/.cache/bazel/_bazel_schroederc/ddace01a1ad38df359eb57115be59fad/kythe/bazel-out/local_linux-fastbuild/bin/kythe/extractors/cmake/compilation_database.runfiles/kythe/extractors/cmake/extract_compilation_database.sh: line 28: jq: command not found
parallel: Warning: $SHELL not set. Using /bin/sh.
parallel: Warning: $HOME not set. Using /tmp
F0423 16:40:41.812670 14979 kindex_tool_main.cc:46] Check failed: in_fd >= 0 (-1 vs. 0) Couldn't open input file /home/schroederc/.cache/bazel/_bazel_schroederc/ddace01a1ad38df359eb57115be59fad/kythe/_tmp/compilation_database_1/d09515d149b3ca237a31caa8fd58f48365e083ce2ebe57173d568d1939fae6b8.kindex
*** Check failure stack trace: ***
               0x4598ad  google::LogMessage::Fail()
               0x458c8c  google::LogMessage::SendToLog()
               0x459563  google::LogMessage::Flush()
               0x45ca88  google::LogMessageFatal::~LogMessageFatal()
               0x408396  DumpIndexFile()
               0x40800b  main
         0x7f8ce7039ec5  __libc_start_main
               0x407e24  (unknown)
                  (nil)  (unknown)
kythe/extractors/cmake/compilation_database: line 30: 14979 Aborted                 "${KINDEX_TOOL}" -suppress_details -explode "${OUT_DIR}/${EXPECTED_INDEX}"
```

I definitely have `jq` installed on my `PATH`.

**Comment by kythe** [comment]
_**schroederc** required changes at 2015-04-23 16:44:58_

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-23 18:50:20_:

  - This one gets me every time (revert grpc BUILD)
  - Hook up jq.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-04-23 18:50:47_:

They're not workarounds--we support Bazel or Campfire; when we run under Bazel, we derive paths from environment variables that Bazel provides; when we're under Campfire, we do it relative to the Campfire root.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-23 18:58:08_:

We don't //want// to support both.  We should have TODOs to help phase out campfire.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-23 19:02:43_:

  - Add a TODO.

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2015-04-23 19:10:43_

**Comment by kythe** [comment]
_**zarko** closed via commit d08a753e9a09b153bbe4732fee00524a26a0d658 at 2015-04-23 19:11:54_
